analytics-api-core/src/main/scala/org/ekstep/analytics/api/service/JobAPIService.scala

```diff
@@ -102,100 +113,66 @@ object JobAPIService {
     }
   }
 
-  private def upsertRequest(body: RequestBody, channel: String)(implicit config: Config): JobRequest = {
-    val outputFormat = body.request.output_format.getOrElse(config.getString("data_exhaust.output_format"))
-    val datasetId = body.request.dataset_id.getOrElse(config.getString("data_exhaust.dataset.default"))
-    val requestId = _getRequestId(body.request.filter.get, outputFormat, datasetId, body.params.get.client_key.get)
-    val job = CassandraUtil.getJobRequest(requestId, body.params.get.client_key.get)
-    val usrReq = body.request
-    val useFilter = usrReq.filter.get
-    val filter = Filter(None, None, None, useFilter.tag, useFilter.tags, useFilter.start_date, useFilter.end_date, useFilter.events, useFilter.app_id, Option(channel))
-    val request = Request(Option(filter), usrReq.summaries, usrReq.trend, usrReq.context, usrReq.query, usrReq.filters, usrReq.config, usrReq.limit, Option(outputFormat), Option(datasetId))
-
-    if (null == job) {
-      _saveJobRequest(requestId, body.params.get.client_key.get, request)
+  private def upsertRequest(body: RequestBody, channel: String)(implicit config: Config, fc: FrameworkContext): JobRequest = {
+    val tag = body.request.tag.getOrElse("")
+    val jobId = body.request.jobId.getOrElse("")
+    val requestedBy = body.request.requestedBy.getOrElse("")
+    val requestId = _getRequestId(tag, jobId, requestedBy, channel)
+    val jobConfig = body.request.jobConfig.getOrElse(Map.empty)
+    val job = postgresDBUtil.getJobRequest(requestId, tag)
+
+    if (job.isEmpty) {
+      _saveJobRequest(requestId, tag, jobId, requestedBy, channel, jobConfig)
     } else {
-      if (StringUtils.equalsIgnoreCase(JobStatus.FAILED.toString(), job.status.get)) {
-        val retryLimit = config.getInt("data_exhaust.retry.limit")
-        val attempts = job.iteration.getOrElse(0)
-        if (attempts < retryLimit) _saveJobRequest(requestId, body.params.get.client_key.get, request, attempts) else job
-      } else job
+      job.get
```

Comment:
The same request can be applied in multiple days to get different exhaust files and this would never create a new request. the logic should be that - if the job is pending then send the same request id or create a new request if the job is completed.

Reply:
Changes for above logic is done